configure.ac

```diff
@@ -3,7 +3,7 @@ AC_PREREQ([2.63])
 AC_INIT(udftools, 2.0, , , [https://github.com/pali/udftools/])
 AC_CONFIG_AUX_DIR(config)
 AM_CONFIG_HEADER(include/config.h:include/config.in)
-AM_INIT_AUTOMAKE
+AM_INIT_AUTOMAKE([subdir-objects])
```

Comment:
It is how autoconf/automake generates Makefiles. More Makefile.am files refers to source files outside of current directory. Without this change, object files (from sources files) are always generated into current directory. With this change, object files are put to same location where are source files. Therefore parallel build starts overwriting generated object files in different directories -- classic race condition.

Reply:
I don't get it sorry. Tons of projects are using subdir-objects with parallel build. subdir-objects will for example prevent object files from overwriting itself if you have foo/foo.c and bar/foo.c. If you have such file and don't use subdir-objects, then foo.o and foo.o will both be produced in the main directory, while with subdir-objects, they wil be produced as foo/foo.o and bar/foo.o.

Reply:
> I don't get it sorry. Tons of projects are using subdir-objects with parallel build.

So if you are proposing some change to merge, I guess you understand what it is doing. But seems not. Argument that tons of project are using it, does not mean that it would work also with udftools in current state.

> subdir-objects will for example prevent object files from overwriting itself if you have foo/foo.c and bar/foo.c.

I do not contradict this. But udftools automake files (Makefile.am) does not use this pattern.

Problem is that e.g. Makefile.am for cdrwtool uses files outside of cdrwtool directory tree. Same for udflabel.

When subdir-objects option is not used, then object files compiled from those source files are put into the **current** main directory as you wrote. Which is the correct behavior, as cdrwtool does not produce any files outside of the cdrwtool directory.

But when subdir-objects is used, then cdrwtool compiles sources files outside of cdrwtool directory. And therefore starts fighting with other Makefile.am files which generates same file in their own directories.

So here **not** usage of subdir-objects is correct way how to avoid above race which you already described.

Reply:
Correct way of course would be to not compile source files outside of the current Makefile.am directory, but that needs fixing Makefile.am automake files, not autoconf file.